aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1585,30 +1614,14 @@ impl AptosVM {
             .finish(Location::Undefined)
     }
 
-    pub(crate) fn make_standard_gas_meter(
-        &self,
-        balance: Gas,
-        log_context: &AdapterLogSchema,
-    ) -> Result<MemoryTrackedGasMeter<StandardGasMeter<StandardGasAlgebra>>, VMStatus> {
-        Ok(MemoryTrackedGasMeter::new(StandardGasMeter::new(
-            StandardGasAlgebra::new(
-                self.gas_feature_version,
-                get_or_vm_startup_failure(&self.gas_params, log_context)?
-                    .vm
-                    .clone(),
-                get_or_vm_startup_failure(&self.storage_gas_params, log_context)?.clone(),
-                balance,
-            ),
-        )))
-    }
-
     fn validate_signed_transaction(
         &self,
         session: &mut SessionExt,
         resolver: &impl AptosMoveResolver,
         transaction: &SignedTransaction,
         transaction_data: &TransactionMetadata,
         log_context: &AdapterLogSchema,
+        is_governance_proposal: bool,
```

Comment:
Can this be part of TransactionMetadata?

Reply:
No, because in order to determine if a transaction is an approved governance proposal, you need to check whether its hash exists in ApprovedExecutionHashes, which is not something you can derive from the transaction itself like other transaction metadata does.

Reply:
All usages of `is_governance_proposal()` function are where we create the metadata, e.g.
```
let txn_metadata = TransactionMetadata::new(txn);
let is_governance_proposal = is_governance_proposal(resolver, txn, &txn_metadata);
``` 
This is true for exactly 2 places where we create metadata. It seems cleaner to have this as a field, instead of passing a flag everywhere, no?

Reply:
Yeah, again I guess the real deciding factor is how we want to define transaction metadata. 
- If we want it to contain info strictly derivable from the transaction itself, then the is_governance_proposal flag has to be passed around
- Otherwise, we may include it in the txn metadata, and deal with the implication of requiring a resolver to compute metadata.

Reply:
Tbh, I am not exactly sure what this metadata is, I think the idea was to have transaction-representation agnostic data structure for the VM, which also caches some information. We already store keyless/randomness related info there: 
```
pub required_deposit: Option<u64>,
pub is_keyless: bool,
```

Reply:
Again, `is_keyless` is information you can derive from the txn itself. `is_governance_proposal` depends on the on-chain state.